packages/loopring_v3/contracts/test/AmmPool.sol

```diff
@@ -231,21 +231,30 @@ contract AmmPool is IBlockReceiver {
         // Withdraw any outstanding balances for the pool account on the exchange
         address[] memory owners = new address[](tokens.length);
         address[] memory tokenAddresses = new address[](tokens.length);
+
         for (uint i = 0; i < tokens.length; i++) {
             owners[i] = address(this);
             tokenAddresses[i] = tokens[i].addr;
         }
         exchange.withdrawFromApprovedWithdrawals(owners, tokenAddresses);
 
         // Withdraw
+        uint96[] memory withdrawn = new uint96[](tokens.length);
         for (uint i = 0; i < tokens.length; i++) {
             address token = tokens[i].addr;
-            require(availableBalance(token, msg.sender) >= amounts[i], "INSUFFICIENT_BALANCE");
-            balance[token][msg.sender] = balance[token][msg.sender].sub(amounts[i]);
-            withdrawInternal(token, amounts[i], msg.sender);
+            uint available = availableBalance(token, msg.sender);
+            if (amounts[i] == 0 || amounts[i] > available) {
```

Comment:
Maybe remove `amounts[i] == 0` here because not withdrawing some tokens may be useful. Just being able to use a very large value to withdraw everything seems enough.